### PR TITLE
gRelayer: Redelivery test & comment fixes

### PR DESCRIPTION
**Pull request by chase-45**

### CODE REVIEW

**Review by derpy-duck — DISMISSED** [review]
Looks good, the only note is I assume we'll probably want three different error codes for clarity rather than combining them all into InvalidOverride

**Comment by derpy-duck** [comment]
Actually quick note: We don't want to revert in the case of an invalid override, we actually want to emit an invalidredelivery event right? Because this is something that could happen due to user error?

File: ethereum/contracts/relayer/coreRelayer/CoreRelayerMessages.sol [hunk 1]
```diff
@@ -344,9 +344,6 @@ contract CoreRelayerMessages is CoreRelayerGetters {
         bool roundUp,
         IRelayProvider provider
     ) internal view returns (uint256 targetAmount) {
-        if (!provider.isChainSupported(targetChain)) {
```
Comment (JoeHowarth) [inline ethereum/contracts/relayer/coreRelayer/CoreRelayerMessages.sol:347]: Are we making the isChainSupported check implicit now? Basically if the src and dst native currency price !== 0 implies it is supported?

**Review by JoeHowarth — APPROVED** [review]
approved w/ comment ^

File: ethereum/contracts/relayer/coreRelayer/CoreRelayerMessages.sol (hunk 1 above)
Comment (derpy-duck) [reply ethereum/contracts/relayer/coreRelayer/CoreRelayerMessages.sol:347]: Oh no, I just moved it to the beginning of 'send/forward/resend'

**Comment by chase-45** [comment]
> Actually quick note: We don't want to revert in the case of an invalid override, we actually want to emit an invalidredelivery event right? Because this is something that could happen due to user error?

I think a revert is more appropriate here, as the InvalidRedelivery can be calculated off-chain just as easily as on chain. Even if the InvalidRedelivery event were posted on-chain, it would not be immediately clear if the Relayer or Integrator was at fault, and we would also want to log an invalid delivery where the VAAs were not properly submitted. Overall I think it's more pragmatic to have invalid deliveries simply not be submitted, as all the error cases can't be detected on-chain.

**Pull request merged**